From 16b9e5fd5cdf4d2dac0b137b0a3aea4b2aa0abcd Mon Sep 17 00:00:00 2001
From: Danny Lin <danny@kdrag0n.dev>
Date: Sun, 13 Dec 2020 18:48:48 -0800
Subject: [PATCH 1/4] wifi: Increase 5 GHz network signal tolerance

On devices with cellular data available, I've been experiencing Wi-Fi
dropouts on 5 GHz networks where it disconnects and falls back to
cellular data around a RSSI of -77 dBm. While the Wi-Fi quality may not
be ideal at this signal level, it is still better to stay on it than
switch to cellular data because switching networks can be very
disruptive to the user.

To make matters worse, the signal tends to oscillate around -77 dBm in
my case, which causes it to oscillate between Wi-Fi and cellular data
every few seconds. This causes far more disruptions than staying on weak
Wi-Fi would.

These signal levels were measured empirically on a Pixel 5, but they
should apply to most devices. 2.4 GHz values were found to be more or
less accurate, but 5 GHz networks continued to work past the AOSP
thresholds. The iPhone 6s was also content with these signal levels and
still displayed 2 of 3 signal levels at -77 dBm.

Signed-off-by: Albert I <kras@raphielgang.org>
---
 .../net/wifi/service/res/values/config.xml    | 24 +++++++++++++++++++
 1 file changed, 24 insertions(+)
 create mode 100644 overlay/frameworks/opt/net/wifi/service/res/values/config.xml

diff --git a/overlay/frameworks/opt/net/wifi/service/res/values/config.xml b/overlay/frameworks/opt/net/wifi/service/res/values/config.xml
new file mode 100644
index 00000000..746f8cea
--- /dev/null
+++ b/overlay/frameworks/opt/net/wifi/service/res/values/config.xml
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+    Copyright (C) 2020 The Proton AOSP Project
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+-->
+<resources>
+    <!-- Integer parameters of the wifi to cellular handover feature
+         wifi should not stick to bad networks -->
+    <!-- Integer threshold for low network score, should be somewhat less than the entry threshhold -->
+    <integer translatable="false" name="config_wifi_framework_wifi_score_bad_rssi_threshold_5GHz">-90</integer>
+    <!-- Integer threshold, do not connect to APs with RSSI lower than the entry threshold -->
+    <integer translatable="false" name="config_wifi_framework_wifi_score_entry_rssi_threshold_5GHz">-85</integer>
+</resources>
-- 
2.30.1


From 11ce495d9ef1f004d8c9dcad47dedf7c7e254c32 Mon Sep 17 00:00:00 2001
From: Albert I <kras@raphielgang.org>
Date: Wed, 17 Feb 2021 01:12:21 +0800
Subject: [PATCH 2/4] packages: Omit SystemUpdates for unofficial builds

Signed-off-by: Albert I <kras@raphielgang.org>
Change-Id: I48eadd5ff8841c56204d76f776d3beab181fe53b
---
 config/packages.mk | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/config/packages.mk b/config/packages.mk
index d620ab23..210cc4f8 100644
--- a/config/packages.mk
+++ b/config/packages.mk
@@ -75,7 +75,11 @@ PRODUCT_DEXPREOPT_SPEED_APPS += \
 
 # Dot Packages
 PRODUCT_PACKAGES += \
-    Customizations \
+    Customizations
+
+ifeq ($(DOT_BUILD_TYPE),OFFICIAL)
+PRODUCT_PACKAGES += \
     SystemUpdates
+endif
 
-include vendor/dot/config/overlay.mk
\ No newline at end of file
+include vendor/dot/config/overlay.mk
-- 
2.30.1


From 82b6ca8a85e90d1d491aa99ab98fe508ae6be300 Mon Sep 17 00:00:00 2001
From: Albert I <kras@raphielgang.org>
Date: Tue, 9 Mar 2021 21:51:30 +0800
Subject: [PATCH 3/4] BoardConfigQcom: Allow targets to upgrade to UM 4.19
 family

Certain sdm660 devices have both received 4.19 kernel port as well as support
for compiling against msm-4.19 HALs. Allow such targets to opt in for being
included as a QSSI supported target.

Signed-off-by: Albert I <kras@raphielgang.org>
Change-Id: Ibcc161641392930d554d78058a6154637c537a34
---
 config/BoardConfigQcom.mk | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/config/BoardConfigQcom.mk b/config/BoardConfigQcom.mk
index 630066ed..dd8c23a4 100644
--- a/config/BoardConfigQcom.mk
+++ b/config/BoardConfigQcom.mk
@@ -18,6 +18,10 @@ UM_4_19_FAMILY := $(KONA) $(LITO) $(BENGAL)
 UM_PLATFORMS := $(UM_3_18_FAMILY) $(UM_4_4_FAMILY) $(UM_4_9_FAMILY) $(UM_4_14_FAMILY) $(UM_4_19_FAMILY)
 QSSI_SUPPORTED_PLATFORMS := $(UM_4_9_FAMILY) $(UM_4_14_FAMILY) $(UM_4_19_FAMILY)
 
+ifeq ($(TARGET_USES_UM_4_19),true)
+    QSSI_SUPPORTED_PLATFORMS += $(TARGET_BOARD_PLATFORM)
+endif
+
 BOARD_USES_ADRENO := true
 
 # UM platforms no longer need this set on O+
-- 
2.30.1


From c88eedac4ac5395a64f79a50db3ea129268f0721 Mon Sep 17 00:00:00 2001
From: Albert I <kras@raphielgang.org>
Date: Mon, 15 Feb 2021 19:05:46 +0800
Subject: [PATCH 4/4] dnm: Inherit KudProject private configuration

Signed-off-by: Albert I <kras@raphielgang.org>
Change-Id: I75e5175e48145499f677c956bc2d3e287fd595cf
---
 config/common.mk | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/config/common.mk b/config/common.mk
index 1ffea8eb..f4381d69 100644
--- a/config/common.mk
+++ b/config/common.mk
@@ -102,6 +102,8 @@ DEVICE_PACKAGE_OVERLAYS += vendor/dot/overlay/common
 #Telephony
 $(call inherit-product, vendor/dot/config/telephony.mk)
 
+-include vendor/kud/config/common.mk
+
 # Packages
 include vendor/dot/config/packages.mk
 
-- 
2.30.1

